### PR TITLE
telegram: add `delete()` when `leave=False`

### PR DESCRIPTION
**Pull request by raulsaavedr**
Add an option for delete bar message with the _**deleteMessage**_ method in Bot API.

### CODE REVIEW

**Comment by raulsaavedr** [comment]
This would be the  behaviour

telegram_delete_method_pull

**Comment by casperdcl** [comment]
Ideally this would be used when `leave=False`?

File: tqdm/contrib/telegram.py [hunk 1]
```diff
@@ -112,6 +122,13 @@ def clear(self, *args, **kwargs):
         if not self.disable:
             self.tgio.write("")
 
+    def close(self):
+        if self.disable:
+            return
+        super(tqdm_telegram, self).close()
+        if not (self.leave or (self.leave is None and self.pos == 0)):
```
Comment (casperdcl) [inline tqdm/contrib/telegram.py:129]: borrowing usual `leave=None|False|True` logic here.

Potential future issue is if people want different behaviour for terminal & telegram outputs (leave one but not the other). They'd have to override the `close()` method.

Comment (raulsaavedr) [reply tqdm/contrib/telegram.py:130]: I think the leave variable is perfect, it fits our needs. And yes in the meantime we don't worry about that case.

**Pull request merged**